cns/fsnotify/fsnotify.go

```diff
@@ -0,0 +1,141 @@
+package fsnotify
+
+import (
+	"context"
+	"io/fs"
+	"os"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type cnsclient interface {
+	ReleaseIPs(ctx context.Context, ipconfig cns.IPConfigsRequest) error
+}
+
+type Watcher struct {
+	CnsClient cnsclient
+}
+
+func WatchFs(w *Watcher, path, directory string, logger *zap.Logger) error {
+	// Create new fs watcher.
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logger.Error("Error creating watcher: ", zap.Error(err))
```

Comment:
this is a structured logger, _you_ don't add kv semantics to the log strings, it does it for you.
this log line is going to look something like:
> {..."message": "Error creating watcher: ", "error": "<errstr>"}  

the message string should just be the string, like:
> > {..."message": "Error creating watcher", "error": "<errstr>"}

```suggestion
		logger.Error("Error creating watcher", zap.Error(err))
```